python_ta/transforms/type_inference_visitor.py

```diff
@@ -465,6 +465,15 @@ def visit_functiondef(self, node: astroid.FunctionDef) -> None:
         func_type = create_Callable(combined_args, combined_return, polymorphic_tvars)
         self.type_constraints.unify(self.lookup_type(node.parent, node.name), func_type)
 
+    def visit_arguments(self, node: astroid.Arguments) -> None:
+        node.inf_type = TypeInfo(NoType)
+        for i in range(len(node.annotations)):
+            if node.annotations[i] is not None:
+                # TODO Create helper function for modifying type_environment and _tvar_to_tnode?
+                tvar = node.parent.type_environment.locals[node.args[i].name]
```

Comment:
This approach is too low-level. Use `lookup_type` and `unify` to express the new constraint for the parameter.